README.md

```diff
@@ -21,4 +21,112 @@ An acme client that integrates with Kubernetes to automatically generate and ren
 
 ## Usage
 
-* _coming soon_
+### Initial setup
+
+At the moment kube-acme is to some extent limited by namespaces, and aspecially the inability
+of ingress to reference service from different namespace. This means that for an ingress
+to be able to properly work, a "local" (as in the "same namespace") kube-acme service is required.
+Still, to properly establish locks, a dedicated "acme" namespace is expected to exist.
+
+```
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: acme
+```
+
+To connect to the acme server a registers user is required (while this will most likely be handled by `kube-acme` in future,
+at the moment a secret has to be created containing all the required information like in example below
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kube-acme-user
+type: Opaque
+data:
+  private.key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1JR2tBZ0VCQkRDNTRZbi9hdU1IMUNFMmJuRmN0ZGE2TnIydFFtbElMeVUzVlA2Nm5pZXA0dUJMRVdxb2IwdmQKMHViNWhOL0pVTCtnQndZRks0RUVBQ0toWkFOaUFBUjFQR1ZvNi9lYy9wR0svMEV3TW9Ud2xwU0Vtcy9USG5VWgpuUWMyUHhXK20wTHhlQlFzZ1hROEJWbVpjRmcrS1E4cHQ4NXpEbjhJWlJwVDNQVXBoeGhucnpiNmJtMzIvWTQwCnhEVWRUL2kxOUp0QzBjRDJyamc0S0w3N1dYdFQraTQ9Ci0tLS0tRU5EIEVDIFBSSVZBVEUgS0VZLS0tLS0K
+  acme-reg.json: ewogICJ0ZXJtc19vZl9zZXJ2aWNlIjogImh0dHBzOi8vbGV0c2VuY3J5cHQub3JnL2RvY3VtZW50cy9MRS1TQS12MS4wLjEtSnVseS0yNy0yMDE1LnBkZiIsCiAgIm5ld19hdXRoenJfdXJpIjogImh0dHBzOi8vYWNtZS1zdGFnaW5nLmFwaS5sZXRzZW5jcnlwdC5vcmcvYWNtZS9uZXctYXV0aHoiLAogICJ1cmkiOiAiaHR0cHM6Ly9hY21lLXN0YWdpbmcuYXBpLmxldHNlbmNyeXB0Lm9yZy9hY21lL3JlZy8xNzg5NzkiLAogICJib2R5IjogewogICAgImFncmVlbWVudCI6ICJodHRwczovL2xldHNlbmNyeXB0Lm9yZy9kb2N1bWVudHMvTEUtU0EtdjEuMC4xLUp1bHktMjctMjAxNS5wZGYiLAogICAgImNvbnRhY3QiOiBbCiAgICAgICJtYWlsdG86ZGVtb0BkZW1vLmNvbSIKICAgIF0sCiAgICAia2V5IjogewogICAgICAieSI6ICJLYmZPY3c1X0NHVWFVOXoxS1ljWVo2ODItbTV0OXYyT05NUTFIVV80dGZTYlF0SEE5cTQ0T0NpLS0xbDdVX291IiwKICAgICAgIngiOiAiZFR4bGFPdjNuUDZSaXY5Qk1ES0U4SmFVaEpyUDB4NTFHWjBITmo4VnZwdEM4WGdVTElGMFBBVlptWEJZUGlrUCIsCiAgICAgICJjcnYiOiAiUC0zODQiLAogICAgICAia3R5IjogIkVDIgogICAgfSwKICAgICJpZCI6IDE3ODk3OSwKICAgICJyZXNvdXJjZSI6ICJyZWciCiAgfQp9Cg==
+```
+
+To generate this, you can use `example/getusersecret.sh -e your@email.com -s acme-staging.api.letsencrypt.org`. 
+This one obviously uses staging letsencrypt servers so you need to change to production one (acme-v01.api.letsencrypt.org) when going prod.
```

Comment:
"In this example, we register with the letsencrypt staging server. In production, you will probably want to use the live acme server: `https://acme-v01.api.letsencrypt.org`